### PR TITLE
Remove deprecated method in ABT public header

### LINKED ISSUE #6543: ABTesting: Remove deprecated APIs
Please remove deprecations in public headers in Firebase 7.

```
~/gh4/firebase-ios-sdk/FirebaseABTesting/Sources/Public/FirebaseABTesting (master) $ git grep -B5 -i deprecat
FIRExperimentController.h-- (void)updateExperimentsWithServiceOrigin:(NSString *)origin
FIRExperimentController.h-                                    events:(FIRLifecycleEvents *)events
FIRExperimentController.h-                                    policy:(ABTExperimentPayloadExperimentOverflowPolicy)policy
FIRExperimentController.h-                             lastStartTime:(NSTimeInterval)lastStartTime
FIRExperimentController.h-                                  payloads:(NSArray<NSData *> *)payloads
FIRExperimentController.h:    DEPRECATED_MSG_ATTRIBUTE("Please use updateExperimentsWithServiceOrigin:events:policy:"
```

### PR DESCRIPTION
**Pull request by christibbs**
Also remove usages in RC, using the new method with completion handler.

Fixes #6543.

### CODE REVIEW

File: FirebaseABTesting/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# v4.2.0
```
Comment (paulb777) [inline FirebaseABTesting/CHANGELOG.md:1]: Will be 7.0.0

**Review by paulb777 — COMMENTED** [review]
Please fix CI.

Change the merge target to firebase7-main or don't merge until after M81 branch is made.

File: FirebaseABTesting/CHANGELOG.md (hunk 1 above)
Comment (christibbs) [reply FirebaseABTesting/CHANGELOG.md]: Resolved in 4cce114.

I changed the base to `firebase7-main` but it ended up pulling in a bunch of cruft into the commits. Did I mess something up?

Comment (paulb777) [reply FirebaseABTesting/CHANGELOG.md]: Yeah, probably the best thing is to rebase your commits onto the firebase7-main branch

**Comment by paulb777** [comment]
this series of commands should be what you need:
```
git fetch
git checkout abt-remove-deprecations
git reset HEAD~ --hard # remove merge commit
git rebase HEAD~~~ --onto origin/firebase7-main # Append last three commits to firebase7-main
git mergetool # Resolve conflict
git push origin abt-remove-deprecations -f  # Push the fixed branch to the origin
```

**Comment by christibbs** [comment]
> this series of commands should be what you need:
> 
> ```
> git fetch
> git checkout abt-remove-deprecations
> git reset HEAD~ --hard # remove merge commit
> git rebase HEAD~~~ --onto origin/firebase7-main # Append last three commits to firebase7-main
> git mergetool # Resolve conflict
> git rebase --continue
> git push origin abt-remove-deprecations -f  # Push the fixed branch to the origin
> ```

Just tried this, when I tried pushing it rejected-- I had to pull and that probably messed things up. Might be easier to just make the change off `firebase7-main` and do a new PR.

**Comment by paulb777** [comment]
OK, or you can schedule something with me if you'd like to pair to learn the git mechanics

**Pull request closed without merging**